docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -274,4 +274,18 @@ An unused bean:
 * does not declare any producer which is eligible for injection to any injection point,
 * is not directly eligible for injection into any `javax.enterprise.inject.Instance` or `javax.inject.Provider` injection point
  
-The extensions can eliminate possible false positives by producing `UnremovableBeanBuildItem`.
\ No newline at end of file
+The extensions can eliminate possible false positives by producing `UnremovableBeanBuildItem`.
+
+Another possibility is to have `quarkus.arc.remove-unused-beans=true` but also set `quarkus.arc.keep-application-beans=true`. In this case beans defined by the application
+will not be removed even if they meet all the criteria set out in the list above.
+This could be useful in cases where bean discovery is needed, like when using
+
+[source,java]
+----
+@Inject
+BeanManager beanManager;
+
+...
+
+beanManager.getBeans(SomeService.class)
```

Comment:
This is more of a question for @mkouba  than a comment on the PR, but if Instance<SomeService> was injected in this case would this be enough to keep the beans from being removed?

Reply:
When I tried it, the manager got injected just fine but no beans were present

Reply:
`@Inject Instance<SomeService>` in app code would prevent the `SomeService` bean from being removed. `@Inject Instance<Object>` effectively disables the "removal" feature completely.